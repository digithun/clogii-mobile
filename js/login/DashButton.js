import React from 'React';
import {Image, View, Text, TextInput, TouchableOpacity} from 'react-native';
import ClogiiButton from 'ClogiiButton';
import styles from './styles';

export default class DashButton extends React.Component {
  render() {
    return (
        <TouchableOpacity onPress={() => this.props.onPress && this.props.onPress()}>
          <Text
            style={{paddingVertical: 40, color: 'white'}}
            >
            ––––––––––– {this.props.caption} ––––––––––
          </Text>
        </TouchableOpacity>
    );
  }
}

export const DashButtonWithContainer = (props) => (<View style={{flex: 1, justifyContent: 'flex-end', alignItems: 'center'}}>
  <DashButton {...props}/>
</View>);
