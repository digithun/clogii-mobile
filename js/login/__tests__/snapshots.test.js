import React from 'react';
import renderer from 'react-test-renderer';

import DashButton from '../DashButton';

describe('DashButton', () => {
  it('render', () => {
    const tree = renderer.create(<DashButton caption="test"/>);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
