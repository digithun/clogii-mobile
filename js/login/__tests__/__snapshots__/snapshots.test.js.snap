exports[`DashButton render 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hitSlop={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "padding": 10,
    }
  }
  testID={undefined}>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 15,
        "marginVertical": 12,
      }
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
      }
    }>
    test
  </Text>
  <View
    style={
      Object {
        "backgroundColor": "rgba(255, 255, 255, 0.5)",
        "flex": 1,
        "height": 1,
        "marginHorizontal": 15,
        "marginVertical": 12,
      }
    } />
</View>
`;
